src/main/scala/mesosphere/marathon/core/instance/Instance.scala

```diff
@@ -124,30 +135,111 @@ object Instance {
       * @param scheduledInstance instance in a Scheduled state
       * @return new instance in a provisioned state
       */
-    def apply(scheduledInstance: Instance, agentInfo: AgentInfo, networkInfo: core.task.state.NetworkInfo, app: AppDefinition, now: Timestamp): Instance = {
+    def apply(
+      scheduledInstance: Instance,
+      agentInfo: AgentInfo,
+      networkInfo: core.task.state.NetworkInfo,
+      app: AppDefinition,
+      now: Timestamp): Instance = {
       require(scheduledInstance.isScheduled, s"Instance '${scheduledInstance.instanceId}' must not be in state '${scheduledInstance.state.condition}'. Scheduled instance is required to create provisioned instance.")
 
-      val task = Task(
-        taskId = Task.Id.forInstanceId(scheduledInstance.instanceId, None),
-        runSpecVersion = app.version,
-        status = Task.Status(
-          stagedAt = now,
-          condition = Condition.Created,
-          networkInfo = networkInfo
-        )
+      val taskIds: Seq[Task.Id] = if (scheduledInstance.hasReservation) {
+        val originalIds = if (scheduledInstance.tasksMap.nonEmpty) {
```

Comment:
I which case can we have a scheduled instance with a non-empty task map?

Reply:
well currently scheduled instance has tasks if it ever got to at least provisioned state, then we keep the task around... before we always created dummy task even for "scheduled" state. Maybe we can go back to doing that so there will always be task when reservation is present...

Reply:
btw. we need this because for every re-launch we need to increment the attempt count, that is part of reserved instance id. so we need to keep tasks around.

Question is if we want to create dummy tasks for scheduled instances or not. I decided not to do that and for that reason, this code exist. Otherwise the id will always exist and we won't need to have an `if` statement here...

Reply:
I would not create dummy tasks. I wonder why we are not removing tasks if an instance goes back to scheduled. The incarnation number is an argument. We might want to track that data somewhere else. E.g. with the instance.